bolts.lic

```diff
@@ -0,0 +1,160 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#bolts
+=end
+
+custom_require.call(%w[common common-items common-crafting])
+
+class Bolts
+  include DRC
+  include DRCC
+  include DRCI
+
+  def initialize
+    @settings = get_settings
+    @bag = @settings.crafting_container
+    @bag_items = @settings.crafting_items_in_container
+    @belt = @settings.engineering_belt
+    @hometown = @settings.hometown
+    @hometown_data = get_data('crafting')['shaping'][@hometown]
+
+    arg_definitions = [
+      [
+        { name: 'type',
+          options: %w[cougar-claw boar-tusk sabertooth angiswaerd hele'la basilisk elsralael soot-stained ice-adder jagged-horn drake-fang],
+          variable: true,
+          descriptions: "Which arrow type to create?" },
+
+        { name: 'lumber',
+          regex: /\w+/i,
+          variable: true,
+          description: 'Type of lumber used.' },
+
+        { name: 'tools',
+          regex: /yes/,
+          optional: true,
+          description: 'Get tools from clerk?'}
+      ],
+      [
+        { name: 'continue', regex: /continue/i, variable: true }
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+
+    if args.continue
+      wait_for_script_to_complete('tinker', ['continue', 'bolts'])
+      exit_tasks
+      exit
+    end
+
+    @type = args.type
+    @lumber = args.lumber
+    @tools = args.tools
+
+    get_tools if @tools
```

Comment:
It's minor, but I'd rather see a guard clause inside `get_tools` and `store_tools`

Reply:
What kind of guard?

Reply:
I added this check/guard: `wait_for_script_to_complete('clerk-tools', ['engineering', 'get']) unless DRCI.exists?('shaper')`

and this `wait_for_script_to_complete('clerk-tools', ['engineering', 'store']) if DRCI.exists?('shaper')`